From 37ec317397e442ee8616b3b2aa81bfb8dde71a06 Mon Sep 17 00:00:00 2001
From: Chet Ramey <chet.ramey@case.edu>
Date: Fri, 17 Nov 2023 16:04:13 -0500
Subject: [PATCH 05/10] Readline-8.2 patch 5: fix for callback signal handling
 when a signal arrives after readline checks but before it restores the
 application's signal handlers

---
 callback.c | 5 ++++-
 patchlevel | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/callback.c b/callback.c
index 69df77d..7209ec3 100644
--- a/callback.c
+++ b/callback.c
@@ -115,7 +115,10 @@ rl_callback_handler_install (const char *prompt, rl_vcpfunc_t *linefunc)
 #define CALLBACK_READ_RETURN() \
   do { \
     if (rl_persistent_signal_handlers == 0) \
-      rl_clear_signals (); \
+      { \
+        rl_clear_signals (); \
+        if (_rl_caught_signal) _rl_signal_handler (_rl_caught_signal); \
+      } \
     return; \
   } while (0)
 #else
diff --git a/patchlevel b/patchlevel
index 626a945..e0ba09d 100644
--- a/patchlevel
+++ b/patchlevel
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-4
+5
-- 
2.38.0.windows.1

